File: examples/canonical-faker/javascript/src/domain.ts
export interface Organization {
  id: string
  name: string
  slug: string
  plan: "free" | "pro" | "enterprise"
  region: "us-east" | "us-west" | "eu-west" | "ap-southeast"
  created_at: string
  owner_email: string
}

export interface User {
  id: string
  email: string
  full_name: string
  locale: "en-US" | "en-GB" | "de-DE" | "fr-FR"
  timezone: string
  created_at: string
}

export interface Membership {
  id: string
  org_id: string
  user_id: string
  role: "owner" | "admin" | "member" | "viewer"
  joined_at: string
  invited_by: string | null
}

export interface Project {
  id: string
  org_id: string
  name: string
  status: "active" | "archived" | "draft"
  created_by: string
  created_at: string
}

export interface AuditEvent {
  id: string
  org_id: string
  user_id: string
  project_id: string
  action: "created" | "updated" | "deleted" | "archived" | "invited" | "exported"
  resource_type: "project" | "user" | "membership" | "api_key" | "invitation"
  resource_id: string
  ts: string
}

export interface ApiKey {
  id: string
  org_id: string
  created_by: string
  name: string
  key_prefix: string
  created_at: string
  last_used_at: string | null
}

export interface Invitation {
  id: string
  org_id: string
  invited_email: string
  role: "admin" | "member" | "viewer"
  invited_by: string
  expires_at: string
  accepted_at: string | null
}

export interface Dataset {
  profileName: string
  seed: number
  organizations: Organization[]
  users: User[]
  memberships: Membership[]
  projects: Project[]
  auditEvents: AuditEvent[]
  apiKeys: ApiKey[]
  invitations: Invitation[]
}

export interface ValidationReport {
  ok: boolean
  violations: string[]
  counts: Record<string, number>
  seed: number
  profileName: string
}

File: examples/canonical-faker/rust/src/domain.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub struct Organization {
    pub id: String,
    pub name: String,
    pub slug: String,
    pub plan: String,
    pub region: String,
    pub created_at: DateTime<Utc>,
    pub owner_email: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub struct User {
    pub id: String,
    pub email: String,
    pub full_name: String,
    pub locale: String,
    pub timezone: String,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub struct Membership {
    pub id: String,
    pub org_id: String,
    pub user_id: String,
    pub role: String,
    pub joined_at: DateTime<Utc>,
    pub invited_by: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub struct Project {
    pub id: String,
    pub org_id: String,
    pub name: String,
    pub status: String,
    pub created_by: String,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub struct AuditEvent {
    pub id: String,
    pub org_id: String,
    pub user_id: String,
    pub project_id: String,
    pub action: String,
    pub resource_type: String,
    pub resource_id: String,
    pub ts: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub struct ApiKey {
    pub id: String,
    pub org_id: String,
    pub created_by: String,
    pub name: String,
    pub key_prefix: String,
    pub created_at: DateTime<Utc>,
    pub last_used_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub struct Invitation {
    pub id: String,
    pub org_id: String,
    pub invited_email: String,
    pub role: String,
    pub invited_by: String,
    pub expires_at: DateTime<Utc>,
    pub accepted_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct Dataset {
    pub profile_name: String,
    pub seed: u64,
    pub organizations: Vec<Organization>,
    pub users: Vec<User>,
    pub memberships: Vec<Membership>,
    pub projects: Vec<Project>,
    pub audit_events: Vec<AuditEvent>,
    pub api_keys: Vec<ApiKey>,
    pub invitations: Vec<Invitation>,
}
